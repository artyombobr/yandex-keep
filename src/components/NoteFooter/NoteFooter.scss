.note {
  &__footer {
    flex-grow: 999;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
  }

  &__info {
    display: flex;
    padding: 0 24px;
    justify-content: space-between;
  }
}

.icons {
  transition: opacity .4s;

  &__item {
    margin-right: 17px;
    cursor: pointer;
    opacity: .5;
    transition: opacity .4s;
    background: none;
    border: none;
    outline: none;

    &:hover{
      opacity: 1;
    }

    & g{
      opacity: inherit;
    }
  }
}

@media screen and (min-width: 768px){
  .icons{
    opacity: 0;
  }

  .note:hover .icons{
    opacity: 1;
  }
}
