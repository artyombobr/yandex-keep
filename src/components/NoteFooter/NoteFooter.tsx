import * as React from "react";
import moment from 'moment';
import 'moment/locale/ru';
import Tags from '../Tags/Tags';

import {ReactComponent as Check} from "../Note/svg/check.svg";
import {ReactComponent as Edit} from "../Note/svg/edit.svg";
import {connect} from "react-redux";


function NoteFooter(props: any) {
  const { note, tags} = props;
  return (
    <div className='note__footer'>
    {tags && <Tags note={note} tags={tags}/>}
    <div className='note__info'>
      <div className='icons'>
        <Check className='icons__item' />
        <Edit className='icons__item' />
      </div>
      {moment(note.created).fromNow()}
    </div>
  </div>
  )
}

const mapStateToProps=(state: any)=>{
  return {
    tags: state.tags
  };
};

export default connect(mapStateToProps)(NoteFooter);
