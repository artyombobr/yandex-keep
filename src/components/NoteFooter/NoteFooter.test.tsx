import React from 'react';
import { shallow } from 'enzyme';
import { NoteFooter } from './NoteFooter';

describe('NoteFooter', () => {
  const note = {
    id: 1,
    type: 'text',
    text: 'Купить чайник с Bluetooth\nВидел у Xiaomi на днях',
    tags: [0],
    size: 's',
    created: 1551593220000,
  };
  const tags = [
    {
      id: 0,
      tag: 'покупки',
    },
  ];

  it('renders correctly', () => {
    const component = shallow(
      <NoteFooter note={note} tags={tags} dispatch={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
