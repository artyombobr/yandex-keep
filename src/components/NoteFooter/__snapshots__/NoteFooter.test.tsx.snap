// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteFooter renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteFooter
    dispatch={[Function]}
    note={
      Object {
        "created": 1551593220000,
        "id": 1,
        "size": "s",
        "tags": Array [
          0,
        ],
        "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
        "type": "text",
      }
    }
    tags={
      Array [
        Object {
          "id": 0,
          "tag": "покупки",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Tags
          note={
            Object {
              "created": 1551593220000,
              "id": 1,
              "size": "s",
              "tags": Array [
                0,
              ],
              "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
              "type": "text",
            }
          }
          tags={
            Array [
              Object {
                "id": 0,
                "tag": "покупки",
              },
            ]
          }
        />,
        <div
          className="note__info"
        >
          <div
            className="icons"
          >
            <button
              className="icons__item"
              onClick={[Function]}
              type="button"
            >
              <ForwardRef />
            </button>
            <button
              className="icons__item"
              type="button"
            >
              <ForwardRef />
            </button>
          </div>
          2 месяца назад
        </div>,
      ],
      "className": "note__footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "note": Object {
            "created": 1551593220000,
            "id": 1,
            "size": "s",
            "tags": Array [
              0,
            ],
            "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
            "type": "text",
          },
          "tags": Array [
            Object {
              "id": 0,
              "tag": "покупки",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="icons"
            >
              <button
                className="icons__item"
                onClick={[Function]}
                type="button"
              >
                <ForwardRef />
              </button>
              <button
                className="icons__item"
                type="button"
              >
                <ForwardRef />
              </button>
            </div>,
            "2 месяца назад",
          ],
          "className": "note__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="icons__item"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef />
                </button>,
                <button
                  className="icons__item"
                  type="button"
                >
                  <ForwardRef />
                </button>,
              ],
              "className": "icons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ForwardRef />,
                  "className": "icons__item",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ForwardRef />,
                  "className": "icons__item",
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": "button",
              },
            ],
            "type": "div",
          },
          "2 месяца назад",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Tags
            note={
              Object {
                "created": 1551593220000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  0,
                ],
                "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
                "type": "text",
              }
            }
            tags={
              Array [
                Object {
                  "id": 0,
                  "tag": "покупки",
                },
              ]
            }
          />,
          <div
            className="note__info"
          >
            <div
              className="icons"
            >
              <button
                className="icons__item"
                onClick={[Function]}
                type="button"
              >
                <ForwardRef />
              </button>
              <button
                className="icons__item"
                type="button"
              >
                <ForwardRef />
              </button>
            </div>
            2 месяца назад
          </div>,
        ],
        "className": "note__footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "note": Object {
              "created": 1551593220000,
              "id": 1,
              "size": "s",
              "tags": Array [
                0,
              ],
              "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
              "type": "text",
            },
            "tags": Array [
              Object {
                "id": 0,
                "tag": "покупки",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="icons"
              >
                <button
                  className="icons__item"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef />
                </button>
                <button
                  className="icons__item"
                  type="button"
                >
                  <ForwardRef />
                </button>
              </div>,
              "2 месяца назад",
            ],
            "className": "note__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="icons__item"
                    onClick={[Function]}
                    type="button"
                  >
                    <ForwardRef />
                  </button>,
                  <button
                    className="icons__item"
                    type="button"
                  >
                    <ForwardRef />
                  </button>,
                ],
                "className": "icons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <ForwardRef />,
                    "className": "icons__item",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <ForwardRef />,
                    "className": "icons__item",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  "type": "button",
                },
              ],
              "type": "div",
            },
            "2 месяца назад",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
